rego/rego.go

```diff
@@ -545,24 +564,127 @@ func (r *Rego) Compile(ctx context.Context) (*CompileResult, error) {
 	return result, nil
 }
 
-func (r *Rego) parse() (map[string]*ast.Module, ast.Body, error) {
+// Prepare will parse inputs, modules, and query arguments in preparation
+// of evaluating them.
+func (r *Rego) Prepare(ctx context.Context) (*PartialResult, error) {
```

Comment:
I'd like to type alias PartialResult to something that indicates it's actual purpose. PartialResult made sense when it held the result of partial evaluation (i.e., the set of conditions that remained after partial evaluation ran) but now that PartialResult is being used to hold all of the context to execute a query it feels like a new name is in order. `rego.PreparedQuery` would be OK, but maybe there's something better we can come up with.

Assuming we use Go 1.9's type aliasing this shouldn't break callers AFAICT.

This doesn't have to block the merge.

Reply:
Cool, I was wondering about that. I'll change the naming on it.